crates/bevy_core_pipeline/src/deferred/node.rs

```diff
@@ -0,0 +1,205 @@
+use bevy_ecs::prelude::*;
+use bevy_ecs::query::QueryItem;
+use bevy_render::render_graph::ViewNode;
+
+use bevy_render::{
+    camera::ExtractedCamera,
+    prelude::Color,
+    render_graph::{NodeRunError, RenderGraphContext},
+    render_phase::RenderPhase,
+    render_resource::{
+        LoadOp, Operations, RenderPassColorAttachment, RenderPassDepthStencilAttachment,
+        RenderPassDescriptor,
+    },
+    renderer::RenderContext,
+    view::ViewDepthTexture,
+};
+#[cfg(feature = "trace")]
+use bevy_utils::tracing::info_span;
+
+use crate::core_3d::{Camera3d, Camera3dDepthLoadOp};
+use crate::prepass::{DepthPrepass, MotionVectorPrepass, NormalPrepass, ViewPrepassTextures};
+
+use super::{AlphaMask3dDeferred, Opaque3dDeferred};
+
+/// Render node used by the prepass.
```

Comment:
```
/// Render node used by the prepass.
```

Is this the GBuffer generation node, or the lighting node? Having trouble figuring out which is where. Imo we should explicitly call them DeferredGBufferPrepass and MainOpaque3dDeferredLightingPass or something.

Reply:
It doesn’t have prepass in the name. It’s the main deferred lighting pass. Maybe we could name it similar to the 3D main pass node but with deferred in the name (and similar node string names) so people understand it is the one of the main passes? @DGriffin91

Reply:
The node @JMS55 is showing here in the code is actually not the lighting node. The lighting node is in `bevy_pbr` and the prepass node is in `bevy_core_pipeline`.

The gbuffer rasterization node is currently called `DeferredNode` (similar to `PrepassNode`)
The lighting node is called `DeferredLightingNode`

I agree that this makes probably sense.